test/common/protobuf/utility_test.cc

```diff
@@ -492,6 +492,7 @@ TEST_F(DeprecatedFieldsTest, NoErrorWhenDeprecatedFieldsUnused) {
   EXPECT_EQ(0, runtime_deprecated_feature_use_.value());
 }
 
+#ifndef ENVOY_DISABLE_DEPRECATED_FEATURES
```

Comment:
Do we expect the number of test need this guard grow or spread when we have more deprecated fields?

If so probably worth adding a macro `DEPRECATED_FEATURE_TEST()` which can be used like:

```
TEST_F(DeprecatedFieldsTest, DEPRECATED_FEATURE_TEST(IndividualFieldDeprecated)) {
  ...
}
```

The macro adds `DISABLED_` prefix to the test name based on `#ifndef ENVOY_DISABLE_DEPRECATED_FEATURES` so the test is excluded from the test run (not compilation).

Reply:
Unclear - this is what we have for the 13 currently deprecated fields, and most off the disabling is actually just for the overall runtime functionality.  I like the idea of the macro though - seems cleaner and easier to see what's disabled.